repo2docker/buildpacks/conda/__init__.py

```diff
@@ -68,10 +70,9 @@ def setup_build_script_files(self):
         files['conda/' + frozen_name] = '/tmp/environment.yml'
         return files
 
-    python_version = Unicode()
-    @default('python_version')
-    def detect_python_version(self):
-        """Detect the Python version for a given environment.yml
+    def get_python_version(self):
```

Comment:
We benefitted from the fact that traitlets cached results (like properties, but only running once). We should be careful to do the same here. For instance, the Python version involves opening and reading a YAML file. That shouldn't happen more than once, so we should make sure that this only runs once, and the computed value is used thereafter.

Reply:
Yeah, I tried to do that but sortof failed (since I was being very mechanical about it), and reverted my changes. I'll give this another shot!

Reply:
Done!